### PR TITLE
[DNM][WIP] feat(1010): use prNum of Job to set SD_PULL_REQUEST

### PR DESCRIPTION
**Pull request by tk3fftk**
## Context
We want to implement `prChain` feature. If `prChain` is true, a job name does not have `PR-` prefix. So we will add `prNum` to Job schema.

## Objective
Use `prNum` instead of a job name.
This PR has backward compatibility so we can also use `PR-` prefix.
This PR also provides some refactoring.

## Misc.
Related Issue: https://github.com/screwdriver-cd/screwdriver/issues/1010
Related PRs:
- https://github.com/screwdriver-cd/data-schema/pull/296
- https://github.com/screwdriver-cd/config-parser/pull/81

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ builds:
       - amd64
     # Include the default settings from https://goreleaser.com/#builds
     # Also include static compilation
-    ldflags: -d -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}} -extldflags "-static"
+    ldflags: -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}} -extldflags "-static"
```
Comment (tk3fftk) [inline .goreleaser.yml]: I don't know it's good to remove `-d` flag but other cli (like `store-cli`) does not have this flag.

File: launch.go [hunk 2]
```diff
@@ -352,12 +353,6 @@ func launch(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, s
 	fmt.Fprintf(emitter, "%s%s\n", blackSprint("Source Dir:     "), w.Src)
 	fmt.Fprintf(emitter, "%s%s\n", blackSprint("Artifacts Dir:  "), w.Artifacts)
 
-	oldJobName := job.Name
-	pr := prNumber(job.Name)
-	if pr != "" {
```
Comment (s-yoshika) [inline launch.go]: Is this if statement no longer necessary?

Comment (tk3fftk) [reply launch.go]: I think it's unnecessary. These are legacy of old workflow era (all pull request jobs are named `main`)

Comment (s-yoshika) [reply launch.go]: 👍

**Pull request closed without merging**